### PR TITLE
fix: redirect issue when creating a post in a different locale and being redirected to the default locale on submit

### PR DESCRIPTION
**Pull request by paulpopus**
fix: redirect issue when creating a post in a different locale and being redirected to the default locale on submit

### CODE REVIEW

File: packages/next/src/views/Edit/index.client.tsx [hunk 1]
```diff
@@ -54,10 +61,12 @@ export const EditViewClient: React.FC = () => {
       getVersions,
       isEditing,
       router,
+      locale,
     ],
   )
 
   useEffect(() => {
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
```
Comment (jacobsfletch) [inline packages/next/src/views/Edit/index.client.tsx:69]: Remove this, add `void` before the promise instead.

File: packages/next/src/views/Edit/index.client.tsx [hunk 2]
```diff
@@ -54,10 +61,12 @@ export const EditViewClient: React.FC = () => {
...
     setOnSave(() => onSave)
```
Comment (jacobsfletch) [inline packages/next/src/views/Edit/index.client.tsx:70]:
```suggestion
    void setOnSave(() => onSave)
```

File: packages/next/src/views/Edit/index.client.tsx [hunk 3]
```diff
@@ -24,19 +25,25 @@ export const EditViewClient: React.FC = () => {
...
+
+  const locale = params.get('locale')
 
   const { Edit, actionsMap } = getComponentMap({
     collectionSlug,
     globalSlug,
   })
 
   const onSave = useCallback(
+    // eslint-disable-next-line @typescript-eslint/require-await
```
Comment (jacobsfletch) [inline packages/next/src/views/Edit/index.client.tsx:38]: Remove this. Instead just remove the `async` keyword.

File: packages/next/src/views/Edit/index.client.tsx [hunk 4]
```diff
@@ -24,19 +25,25 @@ export const EditViewClient: React.FC = () => {
...
     async (json: { doc }) => {
       void getVersions()
       void getDocPermissions()
 
       if (!isEditing) {
-        router.push(`${adminRoute}/collections/${collectionSlug}/${json?.doc?.id}`)
+        const baseRedirectRoute = `${adminRoute}/collections/${collectionSlug}/${json?.doc?.id}`
```
Comment (JarrodMFlesch) [inline packages/next/src/views/Edit/index.client.tsx:44]: You could just write it all inline like it was before, but using the ternary inside the template literal. 

i.e. 
```ts
router.push(`${adminRoute}/collections/${collectionSlug}/${json?.doc?.id}${locale ? `?locale=${locale}` : ''}`)
```

File: packages/next/src/views/Edit/index.client.tsx [hunk 5]
```diff
@@ -24,19 +25,25 @@ export const EditViewClient: React.FC = () => {
...
+        const redirectRoute = locale ? baseRedirectRoute + `?locale=${locale}` : baseRedirectRoute
```
Comment (jacobsfletch) [inline packages/next/src/views/Edit/index.client.tsx:45]: You could make this a one-liner.
```suggestion
        const redirectRoute = `${adminRoute}/collections/${collectionSlug}/${json?.doc?.id}${locale ? `?locale=${locale}` : ''}`
```

File: packages/next/src/views/Edit/index.client.tsx [hunk 6]
```diff
@@ -24,19 +25,25 @@ export const EditViewClient: React.FC = () => {
...
+  const locale = params.get('locale')
 
   const { Edit, actionsMap } = getComponentMap({
     collectionSlug,
     globalSlug,
   })
 
   const onSave = useCallback(
+    // eslint-disable-next-line @typescript-eslint/require-await
     async (json: { doc }) => {
```
Comment (jacobsfletch) [inline packages/next/src/views/Edit/index.client.tsx:39]:
```suggestion
    (json: { doc }) => {
```

**Pull request merged**